### PR TITLE
IGNITE-21808 CREATE ZONE syntax must work with any the case of zone name

### PR DESCRIPTION
**Pull request by JAkutenshi**
https://issues.apache.org/jira/browse/IGNITE-21808

The issue is about changing behavior of `CREATE TABLE ... WITH primary_zone=<zone_name>`. Before my changes it works only as next:
1. Zone is created as quoted identifier: `CREATE ZONE "test_zone"`, then you must use zone name as it defined literal only: `CREATE TABLE test_table (id INT PRIMARY KEY, val INT) WITH PRIMARY_ZONE='test_zone'`.
2. Zone is created as unquoted identifier: `CREATE ZONE test_zone`, then you must use zone name as uppercase literal only: `CREATE TABLE test_table (id INT PRIMARY KEY, val INT) WITH PRIMARY_ZONE='TEST_ZONE'`.

The first case is acceptable as case-sensitive syntax option for some users. But the second case may (and is) confuse a user: zone's name when user creates a zone is identifier, but in table creation statement the name behaves as literal and assumes user's understanding of unclear internal logic: unquoted identifiers are transformed to uppercase implicitly and then, due to `WITH <option_key>=<option_value>` syntax accepts literal only values, there exception is thrown with "Distribution zone with name 'test_zone' not found" message.

The solution is to add an option to use both literal and identifier as option's value. Then, we could execute the next queries without exceptions:
```sql
CREATE ZONE test_zone;
CREATE TABLE test_table (id INT PRIMARY KEY, val INT) WITH PRIMARY_ZONE=test_zone;
```
```sql
CREATE ZONE "test_zone";
CREATE TABLE test_table (id INT PRIMARY KEY, val INT) WITH PRIMARY_ZONE='test_zone';
```
```sql
CREATE ZONE test_zone;
CREATE TABLE test_table (id INT PRIMARY KEY, val INT) WITH PRIMARY_ZONE=`TEST_ZONE`;
```

In the result we're saving previous behaviors (case-sensitive and implicit naming rule usage) and add more clear behavior with name as identifier as well.

---

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

**Comment by JAkutenshi** [comment]
@korlov42 @alievmirza Could you check the request?

File: modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItCreateTableDdlTest.java [hunk 1]
```diff
@@ -370,4 +371,40 @@ public void testPrimaryKeyIndexTypes() {
...
+        sql("DROP TABLE test_table");
+        sql("DROP ZONE test_zone");
+    }
+
+    @Test
+    public void testSuccessfulCreateTableWithZoneQuotedLiteral() {
+        sql("CREATE ZONE \"test_zone\"");
+        sql("CREATE TABLE test_table (id INT PRIMARY KEY, val INT) WITH PRIMARY_ZONE='test_zone'");
+        sql("DROP TABLE test_table");
+        sql("DROP ZONE \"test_zone\"");
```
Comment (alievmirza) [inline modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItCreateTableDdlTest.java:396]: let's drop all created zones in @AfterEach section, as we do for tables. Btw, we don't need to drop table in test

Comment (JAkutenshi) [reply modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItCreateTableDdlTest.java:393]: Done

File: modules/sql-engine/src/main/codegen/includes/parserImpls.ftl [hunk 2]
```diff
@@ -48,7 +48,11 @@ void CreateTableOption(List<SqlNode> list) :
 {
     key = SimpleIdentifier() { s = span(); }
     <EQ>
-    val = Literal()
+    (
+        val = Literal()
+    |
+        val = SimpleIdentifier()
```
[the remark this answers is not in the document]
Comment (JAkutenshi) [reply modules/sql-engine/src/main/codegen/includes/parserImpls.ftl:54]: Done in both test classes

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/ddl/DdlSqlToCommandConverter.java [hunk 3]
```diff
@@ -782,40 +782,74 @@ private String deriveDataStorage(@Nullable SqlIdentifier engineName, PlanningCon
...
+    }
+
+    private static <T, S> T extractValueForUpdateCommandOption(
+            String sqlObjName,
+            Object optId,
+            SqlNode value,
+            DdlOptionInfo<S, T> optInfo,
+            String query
+    ) {
+        var valueKind = value.getKind();
```
[the remark this answers is not in the document]
Comment (JAkutenshi) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/ddl/DdlSqlToCommandConverter.java]: Thank you for noticing. Fixed.

**Review by korlov42 — APPROVED** [review]

**Review by alievmirza — APPROVED** [review]

File: modules/runner/src/testFixtures/java/org/apache/ignite/internal/ClusterPerClassIntegrationTest.java [hunk 4]
```diff
@@ -139,6 +141,19 @@ protected static void dropAllTables() {
         }
     }
 
+    /** Drops all visible tables. */
```
[the remark this answers is not in the document]
Comment (JAkutenshi) [reply modules/runner/src/testFixtures/java/org/apache/ignite/internal/ClusterPerClassIntegrationTest.java]: Nice catch, done!

**Pull request merged**